index.js

```diff
@@ -0,0 +1,120 @@
+const { parseLinks, getPageFromURL, deepMerge } = require('./util');
+
+/**
+ * Helps traverse through paginated resources.
+ */
+class Traverser {
+  /**
+   * Construct a new traverser.
+   *
+   * @param instance An authenticated Axios instance (see NPM module @salling-group/auth).
+   * @param {string} resource The resource to traverse.
+   * @param {Object} baseOptions Axios options to apply to the request.
+   * A mapper to modify the entries in the pages.
+   */
+  constructor(instance, resource, baseOptions = {}) {
+    this.instance = instance;
+    this.resource = resource;
+    this.baseOptions = baseOptions;
+    this.lastPage = null;
+    this.currentPage = 1;
+  }
+
+  /**
+   * Get the current page.
+   *
+   * @returns {Promise<Array>} An array containing the current page.
+   * @throws {Error} If the resource cannot be paginated.
+   */
+  async get() {
+    const result = await this.instance.get(this.resource, deepMerge({ 'params': { 'page': this.currentPage } }, this.baseOptions));
```

Comment:
I think people might be surprised if they pass in `baseOptions` which include the `page` property. What do we want to do in that case?